src/transformers/models/whisper/generation_whisper.py

```diff
@@ -619,6 +633,30 @@ def generate(
             generation_config=generation_config,
         )
 
+        # 5bis speculative decoding: patch generate to make sure the assistant model returns the input tokens
+        # Whisper's generate method does not return the decoder input token ids.
+        # In Transformers, assisted decoding logic does a call to the generate method of the assistant model and expects the decoder input token ids to be returned.
+        # This patch is used to make sure the assistant model returns the decoder input token ids.
+        if "assistant_model" in kwargs:
+            from .modeling_whisper import WhisperForCausalLM
+
+            assistant_model = kwargs["assistant_model"]
+            if assistant_model is self:
+                assistant_model = copy.deepcopy(kwargs["assistant_model"])
+            if (
+                not hasattr(
+                    assistant_model, "_generate_with_input_tokens"
+                )  # make sure the assistant model does not already have the patch
+                and type(assistant_model)
+                != WhisperForCausalLM  # such a patch is not necessary for WhisperForCausalLM that uses directly GenerationMixin's generate
+            ):
+                # patch
```

Comment:
I am guessing we are patching because the assistant model can be another class than `WhisperForCausalLM` right?